### ISSUE TITLE
Add "Other" as a third option for gender

### ISSUE THREAD

**Issue by PierreMesure** [consul/consul#3815]
### User Story

- As a user, I want to have my gender set to "Other", either by the census or manually.
- As an admin, I want to be able to see which proportion of my users are gender non-conforming.

### Why is this Needed?

Although it might not be needed yet in Madrid, more and more countries and cities recognise the possibility of citizens to define themselves as something else than male or female.

### Description

[comment]: # (Feature or Bug? eg. Type: Bug)
*Type*: Feature

I suggest we start by simply adding the category "other" in the database and the user model. A more advanced implementation could follow if the need is expressed.

### Definition of Done
[comment]: # (Any other information that would be useful, bullets are helpful.)

- Add `other` to the model for a user
- Add it in all the statistics page

**Comment by gabrielbaldao** [comment]
Can i contribute with this feature?

**Comment by javierm** [comment]
@gabrielbaldao Sure, go ahead! :smile:

Regarding statistics, one possible implementation would be to show statistics for the genders defined in the database :thinking:. This way they would work on every CONSUL installation, no matter which genders are defined.

**Comment by gabrielbaldao** [comment]
Thanks @javierm, looking here I was wondering about change this 
```scope :male,           -> { where(gender: "male") }``` to [enumerize gem](https://github.com/brainspec/enumerize), what do you think?

**Comment by javierm** [comment]
The enumerize gem looks interesting :smile:. We might add it in the future if it simplifies the code a lot, but for now I'd try to implement this feature without adding a new dependency. We try not to add too many extra dependencies so we don't end up with hundreds of gems in our Gemfile, making it even harder to keep them up to date.

Regarding the places to display the information, maybe we could start by adding an "other" gender to budgets/polls stats when there are users with other genders in the database:

![Stats by gender, showing male and female participation](https://user-images.githubusercontent.com/35156/70658370-287cea00-1c5e-11ea-87a6-6fa7b531e15a.png)

**Comment by gabrielbaldao** [comment]
Right, I agree with this, for this I sayed about do in another pr, but I was having problems to find this page.
Do you suggest some image to show this another gender?

**Comment by javierm** [comment]
@decabeza is the one in charge of icons :smile:. @decabeza any suggestions about an intersex/other icon?

**Comment by javierm** [comment]
@gabrielbaldao @PierreMesure I'd like to ask your opinion about a couple of things I'm considering.

1. Which word should we use? Is "other" OK? Is "nonbinary" better? Or maybe "intersex"? Right now I'd vote for "nonbinary"
2. In the stats, I'd vote for only displaying this gender if there are any participants in this group; it doesn't make much sense to me to display it in CONSUL installations where citizens can only be defined as either "male" or "female" :thinking:

What do you think?
